c/uca/sifter/ucd.cpp

```diff
@@ -0,0 +1,180 @@
+                        nvDecimal,
+                        nvDigit,
+                        nvNumeric,
+                        bm,
+                        unicode1Name,
+                        isoComment,
+                        suc,
+                        slc,
+                        stc] = fields<15>(line);
+            const char32_t codePoint = parseHexCodePoint(cp);
+            auto range = CodePointRange::inclusive(codePoint, codePoint);
+            if (name.ends_with(", First>")) {
+                std::string nextLine;
+                std::getline(unicodeData, nextLine);
+                const auto [lastCp, lastName] = fields<2>(nextLine);
+                CHECK(lastName.ends_with(", Last>")) << line << "\n"
+                                                     << nextLine;
+                CHECK(lastName.starts_with(name.substr(0, name.size() - 8)))
+                        << line << "\n"
+                        << nextLine;
+                CHECK(line.substr(cp.size() + 1 + name.size()) ==
+                      nextLine.substr(lastCp.size() + 1 + lastName.size()))
+                        << line << "\n"
+                        << nextLine;
+                range = CodePointRange::inclusive(codePoint,
+                                                  parseHexCodePoint(lastCp));
+            }
+            coarseGeneralCategory_[gc.front()].addAll(range);
+            generalCategory_[std::string(gc)].addAll(range);
+            for (char32_t c : range) {
+                canonicalCombiningClass_.emplace(c,
+                                                 std::stoi(std::string(ccc)));
+                if (!suc.empty()) {
+                    simpleUppercaseMapping_.emplace(c, parseHexCodePoint(suc));
+                }
+                if (!slc.empty()) {
+                    simpleLowercaseMapping_.emplace(c, parseHexCodePoint(slc));
+                }
+            }
+        }
+    }
+    readMultiPropertyFile(ucdDirectory / "PropList.txt");
+    readMultiPropertyFile(ucdDirectory / "DerivedCoreProperties.txt");
+    {
+        std::ifstream derivedNumericValue(ucdDirectory / "extracted" /
+                                          "DerivedNumericValues.txt");
+        CHECK(derivedNumericValue.good());
+        std::cout << "Reading DerivedNumericValues.txt...\n";
+        for (std::string line; std::getline(derivedNumericValue, line);) {
+            // See https://www.unicode.org/reports/tr44/#UnicodeData.txt.
+            const auto record = stripComment(line);
+            if (record.empty()) {
+                continue;
+            }
+            const auto [cp, decimalNV, _, rationalNV] = fields<4>(record);
+            const auto range = parseHexCodePointRange(cp);
+            codePointsWithNumericValue_.addAll(range);
+            if (rationalNV.find_first_not_of("0123456789") ==
+                std::string_view::npos) {
+                for (const char32_t codePoint : range) {
```

Comment:
```suggestion
            if (rationalNV.find_first_not_of("0123456789") ==
                    std::string_view::npos) {
                for (const char32_t codePoint : range) {
```

Reply:
I really don’t want to fiddle with formatting by hand on this. I’ll see if I can get the .clang-format configuration to do something like that.

Reply:
I got the private:/public: thing to work, but not this.

Reply:
I am confused. the format options claim that they are based on Google style, and Google style is to double the indentation so that the continuation of the if/while/for is indented more than its body and doesn't visually blend in with that.

But this is not a blocker.

Reply:
> Google style is to double the indentation so that the continuation of the if/while/for is indented more than its body and doesn't visually blend in with that.

The style guide has at least one example which aligns things after the ( of an if: https://google.github.io/styleguide/cppguide.html#Boolean_Expressions

Reply:
That example shows the condition-continuation with twice the indent of the if-body. In default Google style, body indent is 2 and condition-continuation is 4. Double those for more conventional indentation, and you get 4 and 8.

Reply:
Right, the example is ambiguous as to whether is indented to the continuation indent or to the (. It appears that the clang-format implementation thinks it is the latter, and I think Google folks maintain that implementation. 🤷.

Reply:
Also note the example below in https://google.github.io/styleguide/cppguide.html#Return_Values.